crypto/rsa/rsa_sp800_56b_gen.c

```diff
@@ -254,32 +253,37 @@ int ossl_rsa_sp800_56b_derive_params_from_pq(RSA *rsa, int nbits,
     if (ossl_rsa_get_lcm(ctx, rsa->p, rsa->q, lcm, gcd, p1, q1, p1q1) != 1)
         goto err;
 
-    /* copy e */
-    BN_free(rsa->e);
-    rsa->e = BN_dup(e);
-    if (rsa->e == NULL)
-        goto err;
+    /*
+     * if e is provided as a parameter, don't recompute e, d or n 
```

Comment:
If you are going to morph a function that was intended for a particular purpose then it needs to be documented correctly..
i.e. the original intention (as described in the params above is that e is passed in). Whilst I understand what you comment means above, it makes little sense in the context of just this function. You are describing how a caller uses this..

Maybe add something to the notes section above.

e.g. 
SP800-56b key generation always passes in a non NULL value for e.
For other purposes, if e is NULL then it is assumed that e, n,and d are already set and do not need to be recalculated.